lib/segment/src/index/hnsw_index/hnsw.rs

```diff
@@ -412,8 +412,13 @@ impl<TGraphLinks: GraphLinks> HNSWIndex<TGraphLinks> {
         config.indexed_vector_count.replace(indexed_vectors);
 
         let graph_links_path = GraphLayers::<TGraphLinks>::get_links_path(path);
-        let graph: GraphLayers<TGraphLinks> =
-            graph_layers_builder.into_graph_layers(Some(&graph_links_path))?;
+        let use_compression =
+            std::env::var_os("__QDRANT_COMPRESSED_LINKS").map_or(false, |x| !x.is_empty());
```

Comment:
Clippy for Rust 1.83 will start suggesting this:

```suggestion
            std::env::var_os("__QDRANT_COMPRESSED_LINKS").is_some_and(|x| !x.is_empty());
```